libyul/optimiser/RedundantAssignEliminator.cpp

```diff
@@ -150,6 +146,7 @@ void RedundantAssignEliminator::operator()(ForLoop const& _forLoop)
 	(*this)(_forLoop.body);
 
 	merge(m_assignments, move(m_forLoopInfo.pendingContinueStmts));
+	m_forLoopInfo.pendingContinueStmts = {};
```

Comment:
Is this line necessary? Even if the vector isn't empty after the move, `m_forLoopInfo.pendingContinueStmts` is never used again and then thrown away.

Reply:
@leonardoalt if you strictly follow the standard, it actually is, as the variable behind the moved value is undefined after the move. That's why it is set to a defined value after the move. Any objections/proposals on how we want to handle such cases?

Reply:
It actually is used, inside `finalize`

Reply:
But then `m_forLoopInfo.pendingBreakStmts` also needs to be cleared after line 157

Reply:
Right!